src/components/FormStep.js

```diff
@@ -135,6 +135,8 @@ const reducer = (draft, action) => {
       draft.canSubmit = canSubmit;
       draft.logicChecking = false;
       draft.isNavigating = false;
+      
+      window.scrollTo({ top: 0, behavior: 'smooth' });
```

Comment:
you should _never_ apply side-effects in state-management code - it's possible that these state updates run multiple times before React even re-renders.

Side-effects like this belong in the `useEffect` hook during rendering, and we have a custom hook for this even: https://github.com/open-formulieren/open-forms-sdk/blob/main/src/hooks/useScrollIntoView.js